### DISCUSSION TITLE
JYers based repository with latest official Marlin bugfix applied

### DISCUSSION THREAD

**Discussion by LCh-77** [Jyers/Marlin#1985] [General Discussion]
Hi everybody, I did a code compilation from Marlin and the latest JYers code and I'am going to make it public for all, you can found the code here: https://github.com/LCh-77/Marlin and you are totally free to use.

I will try to update the code with Marlin latest fixes, now that Marlin doesn't have a official JYERSUI.

**Comment by LCh-77** [comment]


**Comment by tiagofreire-pt** [comment]
@LCh-77 , please, check the code consistency:

```
Compiling .pio\build\STM32F103RE_creality\src\src\lcd\extui\ui_api.cpp.o
Marlin\src\lcd\e3v2\jyersui\dwin.cpp: In static member function 'static void CrealityDWINClass::Menu_Item_Handler(uint8_t, uint8_t, bool)':
Marlin\src\lcd\e3v2\jyersui\dwin.cpp:1545:51: error: 'hotend_info_t' {aka 'struct PIDHeaterInfo<PID_t>'} has no member named 'is_below_target'
 1545 |                 if (thermalManager.temp_hotend[0].is_below_target(-2)) {
      |                                                   ^~~~~~~~~~~~~~~
Marlin\src\lcd\e3v2\jyersui\dwin.cpp:1952:51: error: 'hotend_info_t' {aka 'struct PIDHeaterInfo<PID_t>'} has no member named 'is_below_target'
 1952 |                 if (thermalManager.temp_hotend[0].is_below_target(-2)) {
      |                                                   ^~~~~~~~~~~~~~~
Marlin\src\lcd\e3v2\jyersui\dwin.cpp:1971:51: error: 'hotend_info_t' {aka 'struct PIDHeaterInfo<PID_t>'} has no member named 'is_below_target'
      |                                                   ^~~~~~~~~~~~~~~
Marlin\src\lcd\e3v2\jyersui\dwin.cpp:1989:51: error: 'hotend_info_t' {aka 'struct PIDHeaterInfo<PID_t>'} has no member named 'is_below_target'
 1989 |                 if (thermalManager.temp_hotend[0].is_below_target(-2)) {
      |                                                   ^~~~~~~~~~~~~~~
Marlin\src\lcd\e3v2\jyersui\dwin.cpp: In static member function 'static void CrealityDWINClass::Popup_Control()':
Marlin\src\lcd\e3v2\jyersui\dwin.cpp:5783:49: error: 'hotend_info_t' {aka 'struct PIDHeaterInfo<PID_t>'} has no member named 'is_below_target'
 5783 |               if (thermalManager.temp_hotend[0].is_below_target(-2)) {
      |                                                 ^~~~~~~~~~~~~~~
*** [.pio\build\STM32F103RE_creality\src\src\lcd\e3v2\jyersui\dwin.cpp.o] Error 1
```

**Reply by tiagofreire-pt** [reply]
I'm trying to be agnostic to this mess of "who-is-copying-who".

Just trying to get a firmware with some issues solved definitely, such as https://github.com/MarlinFirmware/Marlin/issues/23700.

If you would consider merging yours and his code into one single repo, **crediting at the front page the one's work**, would be more beneficial to the community, with no prejudice for eventual legal enforcement on previous copyright infringements. After all, the code is all under GPL v3.

**Reply by tiagofreire-pt** [reply]
Is this one the most cutting edge? https://github.com/tititopher68-dev2/Marlin/tree/new_feature

**Reply by tiagofreire-pt** [reply]
Where can I find a full fix for this bug? https://github.com/MarlinFirmware/Marlin/issues/23700

**Reply by LCh-77** [reply]
????

**Reply by tiagofreire-pt** [reply]
> ????

https://github.com/MarlinFirmware/Marlin/issues/23700#issuecomment-1169667295

**Reply by tiagofreire-pt** [reply]
Hey, both of you, stop! If you have anything to say to each other, just take it to a court of law!

**Reply by LCh-77** [reply]
> > ????
> 
> [MarlinFirmware#23700 (comment)](https://github.com/MarlinFirmware/Marlin/issues/23700#issuecomment-1169667295)

Test it now.

**Comment by FunkyNuggets007** [comment]
I'm glad I have a different printer now. This is ridiculous

**Reply by alphatrianguli** [reply]
Always! @tititopher68-dev, I wrote to you on the Discord btw. with a question on the 3.0.0 FW

**Comment by LCh-77** [comment]
Updated the ENHANCED branch, now compile without errors.

**Comment by LCh-77** [comment]


**Comment by CRCinAU** [comment]
Please don't tag me in threads about this bullshit anymore. I'm sick of the constant arguing and toxic behaviour.

This thread highlights exactly what the problem is, and why JyersUI was pulled from Marlin.

**Comment by LCh-77** [comment]
The author of this firmware is JYers! Everyone else are contributors.

![ttf](https://user-images.githubusercontent.com/98503100/178649231-485a456e-dbf6-40f5-8a0c-2af1fe36318d.jpg)

![ttf2](https://user-images.githubusercontent.com/98503100/178649249-9964b383-d806-407f-92cc-6f57f7779036.jpg)

![ttf3](https://user-images.githubusercontent.com/98503100/188802653-16646b8a-3d75-4587-b000-d3744f2e0017.jpg)

**Comment by LCh-77** [comment]
I  posted a release: https://github.com/LCh-77/Marlin/releases/tag/BLTUBL-LA
Also now the Enhanced branch comes with working configuration files.
